engine/src/main/java/io/camunda/zeebe/engine/state/instance/DbJobState.java

```diff
@@ -309,14 +320,30 @@ public boolean isInState(final long key, final State state) {
 
   @Override
   public void forEachActivatableJobs(
-      final DirectBuffer type, final BiFunction<Long, JobRecord, Boolean> callback) {
+      final DirectBuffer type,
+      final List<String> tenantIds,
+      final BiFunction<Long, JobRecord, Boolean> callback) {
     jobTypeKey.wrapBuffer(type);
 
     activatableColumnFamily.whileEqualPrefix(
         jobTypeKey,
-        ((compositeKey, zbNil) -> {
-          final long jobKey = compositeKey.second().inner().getValue();
-          return visitJob(jobKey, callback::apply);
+        ((tenantAwareCompositeKey, zbNil) -> {
+          final DbLong jobKey = tenantAwareCompositeKey.first().second().inner();
+
+          String tenantId = TenantOwned.DEFAULT_TENANT_IDENTIFIER;
+          if (tenantAwareCompositeKey.getLength()
+              > jobTypeKey.getLength()
+                  + jobKey.getLength()
+                  + 4) { // account 4 bytes for an empty string reference (for pre-tenant-aware
+            // data)
```

Comment:
🔧 I think we should put this logic inside of the `DbTenantAwareKey`. We can add some getter that gets the tenantId as a String. This can be clever and say that if the type is SUFFIX and we don't actually have a tenant prepended it can return the default.